scipy/signal/_signaltools.py

```diff
@@ -3887,6 +3887,37 @@ def resample_poly(x, up, down, axis=0, window=('kaiser', 5.0),
     return y_keep
 
 
+def _angle(z, xp):
+    """np.angle replacement
+    """
+    # XXX: https://github.com/data-apis/array-api/issues/595
+    if xp.isdtype(z.dtype, 'complex floating'):
+        zreal, zimag = xp.real(z), xp.imag(z)
+    else:
+        zreal, zimag = z, 0
+
+    a = xp.atan2(zimag, zreal)
+    return a
+
+
+def _skip_if_float(arg):
+    # XXX: remove when https://github.com/data-apis/array-api-compat/pull/147
```

Comment:
Looks like this has now been merged

Reply:
This will actually need some small work on the scipy side: we call `compliance_scipy` which chokes on ints/floats etc.